File: MaterialLibrary/src/main/java/com/mchapagai/library/utils/DateTimeUtils.java
package com.mchapagai.library.utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public class DateTimeUtils {

    private static SimpleDateFormat simpleDateMonthNameFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
    private static SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.US);

    /**
     * Helper method for date utilities
     *
     * @param s date string value
     * @return the date in May, 12, 2018 format
     */
    public static String getNameOfMonth(String s) {
        SimpleDateFormat nameOfMonthFormat = new SimpleDateFormat("MMM dd, yyy", Locale.US);
        Date date = null;
        try {
            date = simpleDateMonthNameFormat.parse(s);
        } catch (ParseException e) {
            e.getMessage();
        }

        return nameOfMonthFormat.format(date);
    }

    /**
     * Helper method for date utilities
     *
     * @param s date string value
     * @return the date in 2018 format
     */
    public static String getYearOnly(String s) {
        SimpleDateFormat yearFormat = new SimpleDateFormat("yyyy", Locale.US);
        Date date = null;
        try {
            date = simpleDateFormat.parse(s);
        } catch (ParseException e) {
            e.getMessage();
        }

        return yearFormat.format(date);
    }

    public static Calendar getCalender(String timestamp, String pattern) {
        Calendar calendar = Calendar.getInstance();
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern, Locale.US);
        simpleDateFormat.setTimeZone(TimeZone.getDefault());
        try {
            Date date = simpleDateFormat.parse(timestamp);
            calendar.setTime(date);
        } catch (ParseException e) {
            e.getMessage();
        }
        return calendar;
    }

}

File: Core/src/main/java/com/mchapagai/core/utils/DateTimeUtils.kt
package com.mchapagai.core.utils

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale
import java.util.TimeZone

object DateTimeUtils {

    private const val DATE_PATTERN_MONTH_NAME = "MMM dd, yyy"
    private const val DATE_PATTERN_YEAR_ONLY = "yyyy"
    private const val DATE_PATTERN_DEFAULT = "yyyy-MM-dd"

    fun getNameOfMonth(dateString: String): String {
        val inputFormat = SimpleDateFormat(DATE_PATTERN_DEFAULT, Locale.US)
        val outputFormat = SimpleDateFormat(DATE_PATTERN_MONTH_NAME, Locale.US)
        return try {
            val date = inputFormat.parse(dateString)
            outputFormat.format(date)
        } catch (e: Exception) {
            e.localizedMessage
            ""
        }
    }

    fun getYearOnly(dateString: String): String {
        val inputFormat = SimpleDateFormat(DATE_PATTERN_DEFAULT, Locale.US)
        val outputFormat = SimpleDateFormat(DATE_PATTERN_YEAR_ONLY, Locale.US)
        return try {
            val date = inputFormat.parse(dateString)
            outputFormat.format(date)
        } catch (e: Exception) {
            e.localizedMessage
            ""
        }
    }

    fun getCalendar(timestamp: String, pattern: String): Calendar {
        val calendar = Calendar.getInstance()
        val simpleDateFormat = SimpleDateFormat(pattern, Locale.US)
        simpleDateFormat.timeZone = TimeZone.getDefault()
        try {
            val date = simpleDateFormat.parse(timestamp)
            calendar.time = date
        } catch (e: Exception) {
            e.localizedMessage
        }
        return calendar
    }
}